src/array_api_stubs/_draft/searching_functions.py

```diff
@@ -139,21 +139,33 @@ def searchsorted(
     """
 
 
-def where(condition: array, x1: array, x2: array, /) -> array:
+def where(
+    condition: array,
```

Comment:
Might be better to require that condition is an array ISTM. `x1` and `x2` can be arrays or scalars, but there should be at least one array, and `condition` it is.

```
In [7]: torch.where(True, 3, 4)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[7], line 1
----> 1 torch.where(True, 3, 4)

TypeError: where() received an invalid combination of arguments - got (bool, int, int), but expected one of:
 * (Tensor condition)
 * (Tensor condition, Tensor input, Tensor other, *, Tensor out = None)
 * (Tensor condition, Number self, Tensor other)
      didn't match because some of the arguments have invalid types: (bool, int, int)
 * (Tensor condition, Tensor input, Number other)
      didn't match because some of the arguments have invalid types: (bool, int, int)
 * (Tensor condition, Number self, Number other)
      didn't match because some of the arguments have invalid types: (bool, int, int)

In [2]: cupy.where(True, 3, 4)
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[2], line 1
----> 1 cupy.where(True, 3, 4)

...

AttributeError: 'bool' object has no attribute 'astype'
```

Reply:
Yes, there should be at least one array, but it is not clear to me that `condition` should be required to be an array. While Torch does not support a scalar condition, NumPy does.

```python
In [8]: np.where(True,np.ones((3,3)),0)
Out[8]:
array([[1., 1., 1.],
       [1., 1., 1.],
       [1., 1., 1.]])
```

It seems odd to me to be as restrictive as PyTorch when PyTorch supports

```python
In [9]: torch.where(torch.asarray(True),torch.ones((3,3)),0)
Out[9]:
tensor([[1., 1., 1.],
        [1., 1., 1.],
        [1., 1., 1.]])
```

Reply:
From my (non-extensive) checks, the variant "condition is an array, x1 and x2 are arrays or scalars" covers all major array libraries. And it seems to be the most straightforward to explain in words or legalese, and most useful downstream, FWIW. 

The `In [9]` variant above seems to be covered, too? As long as the algorithm is 
- convert x1 and x2 to arrays
- broadcast all of `condition, x1, x2`
- profit!

Reply:
Yes, the point I was trying to convey was that 

```
torch.where(torch.asarray(True),torch.ones((3,3)),0)
```

can be considered sugar for

```
torch.where(True,torch.ones((3,3)),0)
```

Not clear, from a standards perspective, why the latter should be prohibited a priori, as in PyTorch.

Reply:
as per https://github.com/data-apis/array-api/pull/860#issuecomment-2541637555 I think this is fine as is